engine/script/src/script_http_util.h

```diff
@@ -22,13 +54,23 @@ namespace dmScript
 
         lua_newtable(L);
 
-        lua_pushliteral(L, "status");
         lua_pushinteger(L, resp->m_Status);
-        lua_rawset(L, -3);
+        lua_setfield(L, -2, "status");
 
-        lua_pushliteral(L, "response");
-        lua_pushlstring(L, response, resp->m_ResponseLength);
-        lua_rawset(L, -3);
+        if (resp->m_Path)
+        {
+            if (!WriteResponseToFile(resp->m_Path, response, resp->m_ResponseLength))
+            {
+                lua_pushstring(L, "Failed to write to temp file");
+                lua_setfield(L, -2, "error");
+            }
+
+            lua_pushstring(L, resp->m_Path);
+            lua_setfield(L, -2, "response");
```

Comment:
The response doubles as payload or path

Reply:
We could also do `{ response = nil, path = path }`

Reply:
I prefer using path instead of response. If you have a generic http.request() response handler function it does not necessarily know if a response was written to disk or returned as a string.